app/validators/routes_and_redirects_validator.rb

```diff
@@ -113,76 +113,78 @@ def additional_keys(route, attribute)
   end
 
   class RedirectValidator
-    def validate(record, redirect)
+    attr_reader :redirect, :errors
+
+    def initialize(record, redirect)
+      @redirect = redirect
+      @errors = record.errors[:redirects]
+    end
+
+    def validate
       path = redirect[:path]
       destination = redirect[:destination]
-      type = redirect[:type]
 
-      unless path.present?
-        record.errors[:redirects] << "path must be present"
-      end
+      errors << "path must be present" unless path.present?
+      errors << "destination must be present" unless destination.present?
 
-      unless destination.present?
-        record.errors[:redirects] << "destination must be present"
-      end
+      return unless path.present? && destination.present?
 
-      if path == destination
-        record.errors[:redirects] << "path cannot equal the destination"
-      end
+      errors << "path cannot equal the destination" if path == destination
 
-      unless valid_exact_redirect_target?(destination)
-        record.errors[:redirects] << "is not a valid redirect destination"
+      begin
+        uri = URI.parse(destination)
+      rescue URI::InvalidURIError
+        errors << "destination is an invalid url"
+        return
       end
+
+      return if destination.starts_with?("/")
```

Comment:
it is better to avoid a `return` in the middle of a method:

```ruby
    validate_external_redirect(destination, uri) if external_redirect?
  end

private

  def external_redirect?(destination)
    !destination.starts_with?("/")
  end

```

By extracting the method it also reads a bit better